src/syntax.c

```diff
@@ -2584,6 +2584,12 @@ update_si_attr(int idx)
 		sip->si_flags |= HL_TRANS_CONT;
 		sip->si_cont_list = CUR_STATE(idx - 1).si_cont_list;
 	    }
+		if (spp->sp_type == ST_REG && spp->sp_end_pattern != NULL &&
+			sip->si_attr ==
+				syn_get_match_attr(spp->sp_end_pattern, 'matchgroup'))
+	    {
+		sip->si_attr = CUR_STATE(idx - 1).si_attr;
+	    }
```

Comment:
Coding style issue:  For a single statement in a `if`, curly braces are not used.